tools/lint/lib/checks/esid.py

```diff
@@ -30,7 +30,15 @@ def run(self, name, meta, source):
 
         if 'esid' in meta:
             esid = str(meta['esid'])
-            if self.esidRegex.match(esid) == None:
+
+            try:
+                parsed = urlparse('#%s' % esid)
+
+                if parsed.fragment == esid:
+                    return
+                else:
+                    raise ValueError()
+            except ValueError:
```

Comment:
for consistency with the other returns, how about:

```python
if 'esid' in meta:
    esid = str(meta['esid'])
    parsed = urlparse('#%s' % esid)
    if parsed.fragment != esid:
        return 'The `esid` tag has the wrong format: %s' % esid
```

Reply:
But https://github.com/tc39/test262/blob/master/tools/lint/lib/checks/license.py#L37-L43 ?

Reply:
this is just wrapping an error raised on line 40, not forwarding the error to outside the of this function call.

At the end, this try catch block will only serve to return the string anyway.

Reply:
unless you tell me `urlparse` may also throw a ValueError. In that case this code would be valid.